### PR TITLE
evm: Echidna stateful fuzzing for NttManager

### PR DESCRIPTION
**Pull request by djb15**
This PR adds echidna stateful fuzzing for the manager and performs a fuzzing run as part of the CI pipeline now.

The fuzzing suite only touches the manager due to time constraints.

### CODE REVIEW

File: .github/workflows/evm.yml [hunk 1]
```diff
@@ -41,3 +41,32 @@ jobs:
...
+
+      - name: Run Forge build
+        run: |
+          make test-push0
+        id: build
+
+      - name: Install Echidna
+        run: |
+          curl -LO https://github.com/crytic/echidna/releases/download/v2.2.3/echidna-2.2.3-x86_64-linux.tar.gz
+          tar -xzf echidna-2.2.3-x86_64-linux.tar.gz
```
Comment (johnsaigle) [inline .github/workflows/evm.yml:66]: Maybe we can use `xz` instead? I hear it has great new features

Comment (johnsaigle) [reply .github/workflows/evm.yml:66]: (this is a joke)

**Review by johnsaigle — APPROVED** [review]
Happy to have echidna in the test suite. I've not used it before but this doesn't touch any production code so I don't see any harm in us merging it.

**Review by RahulMaganti47 — APPROVED** [review]
really nice to have! no impact on production code - im ok with merging

**Pull request merged**